pages/ref/api-debug.mdx

```diff
@@ -0,0 +1,190 @@
+
+Usage example:
+
+```tact
+// now() has to return a value greater than 1000, otherwise an error message will be thrown
+require(now() > 1000, "We're in the first 1000 seconds of 1 January 1970!");
+```
+
+## dump
+
+```tact
+fun dump(arg);
+```
+
+Prints the argument `arg` to the contract's debug console. Evaluated only if `debug` option is set in the [configuration file](/book/config), otherwise does nothing.
+
+Can be applied to the following list of types and values:
+
+* [`Int{:tact}`][int]
+* [`Bool{:tact}`][bool]
+* [`Address{:tact}`][p]
+* [`Builder{:tact}`][p], [`Cell{:tact}`][p] or [`Slice{:tact}`][p]
+* [`String{:tact}`][p] or [`StringBuilder{:tact}`][p]
+* [`map<K, V>{:tact}`](/book/maps)
+* [Optionals and `null{:tact}` value](/book/optionals)
+* `void`, which is implicitly returned when a function doesn't have return value defined
+
+Usage examples:
+
+```tact
+// Int
+dump(42);
+
+// Bool
+dump(true);
+dump(false);
+
+// Address
+dump(myAddress());
+
+// Builder, Cell or Slice
+dump(beginCell());           // Builder
+dump(emptyCell());           // Cell
+dump(emptyCell().asSlice()); // Slice
+
+// String or StringBuilder
+dump("Hello, my name is..."); // String
+dump(beginTailString());      // StringBuilder
+
+// Maps
+let m: map<Int, Int> = emptyMap();
+m.set(2 + 2, 4);
+dump(m);
+
+// Special values
+dump(null);
+dump(emit("msg".asComment())); // As emit() function doesn't return a value, dump() would print #DEBUG#: void.
+```
+
+## dumpStack
+
+```tact
+fun dumpStack();
+```
+
+Prints all the values of [persistent state variables](/book/contracts#variables) to the contract's debug console. Evaluated only if `debug` option is set in the [configuration file](/book/config), otherwise does nothing.
+
+Usage example:
+
+```tact
+contract DumpsterFire {
+    var1: Int = 0;
+    var2: Int = 5;
+
+    receive() {
+        dumpStack(); // would print 0 5
+    }
+}
+```
+
+## throw
+
+```tact
+fun throw(code: Int);
+```
+
+An alias to [`nativeThrow(){:tact}`](#nativethrow).
+
+## nativeThrow
+
+```tact
+fun nativeThrow(code: Int);
+````
+
+Throws an exception with an error code equal to `code`. Execution of the current context stops (the statements after `nativeThrow` won't be executed) and control will be passed to the first [`try...catch{:tact}` block](/book/statements#try-catch) in the call stack. If no `try{:tact}` or `try...catch{:tact}` block exists among caller functions, [TVM](https://docs.ton.org/learn/tvm-instructions/tvm-overview) will terminate the transaction.
+
+Usage examples:
+
+```tact {2,7}
+fun thisWillTerminate() {
+    nativeThrow(42); // throwing with exit code 42
+}
+
+fun butThisDoesnt() {
+    try {
+        nativeThrow(42); // throwing with exit code 42
+    }
+
+    // ... follow-up logic ...
+}
+```
+
+<Callout>
+
+  Prefer to use `require(){:tact}` over this function as the former comes with more [compile-time](/ref/api-comptime) optimizations and is generally more user-friendly.
```

Comment:
`require` is not always user-friendly as it generates its own exit codes which are not immediately obvious from Tact code

I think we should actually deprecate `require` just because of this

Reply:
> I think we should actually deprecate `require` just because of this

I don't think that's necessary. We could try to make `require` always throw a certain error code, like 138 (not used yet)